spectrocloud/resource_cluster_openstack.go

```diff
@@ -523,56 +533,62 @@ func resourceClusterOpenStackUpdate(ctx context.Context, d *schema.ResourceData,
-		for _, mp := range os {
-			machinePool := mp.(map[string]interface{})
-			osMap[machinePool["name"].(string)] = machinePool
+		for _, mp := range os.List() {
+			machinePoolResource := mp.(map[string]interface{})
+			name := machinePoolResource["name"].(string)
+			if name != "" {
+				osMap[name] = machinePoolResource
+			}
 		}
 		nsMap := make(map[string]interface{})
-		for _, mp := range ns {
+		for _, mp := range ns.List() {
 			machinePoolResource := mp.(map[string]interface{})
-			nsMap[machinePoolResource["name"].(string)] = machinePoolResource
-			// since known issue in TF SDK: https://github.com/hashicorp/terraform-plugin-sdk/issues/588
-			if machinePoolResource["name"].(string) != "" {
-				name := machinePoolResource["name"].(string)
-				hash := resourceMachinePoolOpenStackHash(machinePoolResource)
-
-				var err error
-				machinePool, err := toMachinePoolOpenStack(machinePoolResource)
-				if err != nil {
-					return diag.FromErr(err)
-				}
-
-				if oldMachinePool, ok := osMap[name]; !ok {
-					log.Printf("Create machine pool %s", name)
-					err = c.CreateMachinePoolOpenStack(cloudConfigId, machinePool)
-				} else if hash != resourceMachinePoolOpenStackHash(oldMachinePool) {
-					log.Printf("Change in machine pool %s", name)
-					err = c.UpdateMachinePoolOpenStack(cloudConfigId, machinePool)
-					// Node Maintenance Actions
-					err := resourceNodeAction(c, ctx, nsMap[name], c.GetNodeMaintenanceStatusOpenStack, CloudConfig.Kind, cloudConfigId, name)
+			name := machinePoolResource["name"].(string)
+			if name != "" {
+				nsMap[name] = machinePoolResource
+				if oldMachinePool, exists := osMap[name]; !exists {
+					log.Printf("[DEBUG] Create machine pool %s", name)
+					machinePool, err := toMachinePoolOpenStack(machinePoolResource)
 					if err != nil {
 						return diag.FromErr(err)
 					}
+					if err := c.CreateMachinePoolOpenStack(cloudConfigId, machinePool); err != nil {
+						return diag.FromErr(err)
+					}
+				} else {
+					oldHash := resourceMachinePoolOpenStackHash(oldMachinePool)
+					newHash := resourceMachinePoolOpenStackHash(machinePoolResource)
+
+					if oldHash != newHash {
+						log.Printf("[DEBUG] Updating machine pool %s (hash changed: %d -> %d)", name, oldHash, newHash)
+						machinePool, err := toMachinePoolOpenStack(machinePoolResource)
+						if err != nil {
+							return diag.FromErr(err)
+						}
+						if err := c.UpdateMachinePoolOpenStack(cloudConfigId, machinePool); err != nil {
+							return diag.FromErr(err)
+						}
+						err = resourceNodeAction(c, ctx, machinePoolResource, c.GetNodeMaintenanceStatusOpenStack, CloudConfig.Kind, cloudConfigId, name)
+						if err != nil {
+							return diag.FromErr(err)
+						}
+					} else {
+						log.Printf("[DEBUG] Machine pool %s unchanged (hash: %d)", name, oldHash)
+					}
 				}
-
-				if err != nil {
-					return diag.FromErr(err)
-				}
-
-				// Processed (if exists)
 				delete(osMap, name)
+			} else {
```

Comment:
@manimaran89 remove this else part if nome is not specified huubble should through error